### ISSUE TITLE
Can't start nx nextjs inside Docker with pm2

### ISSUE THREAD

**Issue by vendramini** [nrwl/nx#28313]
### Current Behavior

I'm trying to add a nginx in front of my nextjs app using Docker.

NX Dockerfile
```
FROM node:alpine3.19

# Set working directory
WORKDIR /usr/app

# Install PM2 globally
RUN npm install --global nx@19.8.4 pm2

# Copy package.json and package-lock.json before other files
# Utilise Docker cache to save re-installing dependencies if unchanged
COPY ./package*.json ./

# Install dependencies
RUN npm install --force

# Copy all files
COPY ./ ./

#ENV NX_DAEMON=false
#ENV NX_SKIP_NX_CACHE=true
#ENV NX_CACHE_PROJECT_GRAPH=false

# Build app
RUN nx build docker-test --verbose

# Expose the listening port
EXPOSE 3000

# Run container as non-root (unprivileged) user
# The node user is provided in the Node.js Alpine base image
USER node

# Run npm start script with PM2 when container starts
CMD [ "pm2-runtime", "nx", "--", "start", "docker-test" ]
```

nginx Dockerfile
```
# Base on offical NGINX Alpine image
FROM nginx:alpine

# Remove any existing config files
RUN rm /etc/nginx/conf.d/*

# Copy config files
# *.conf files in conf.d/ dir get included in main config
COPY ./default.conf /etc/nginx/conf.d/

# Expose the listening port
EXPOSE 80

# Launch NGINX
CMD [ "nginx", "-g", "daemon off;" ]
```

docker-compose.yml
```
services:
  nextjs:
    build: ./
  nginx:
    build: ./nginx
    ports:
      - 80:80
```

Docker console
```
nextjs-1  | (node:1136) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
nextjs-1  | (Use `node --trace-deprecation ...` to show where the warning was created)
nextjs-1  |  NX   Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
nextjs-1  | Pass --verbose to see the stacktraces.
nextjs-1  | 2024-10-05T00:28:04: PM2 log: App [nx:0] exited with code [1] via signal [SIGINT]
nextjs-1  | 2024-10-05T00:28:04: PM2 log: App [nx:0] starting in -fork mode-
nextjs-1  | 2024-10-05T00:28:04: PM2 log: App [nx:0] online
```

I tried playing with 
```
#ENV NX_DAEMON=false 
#ENV NX_SKIP_NX_CACHE=true 
#ENV NX_CACHE_PROJECT_GRAPH=false 
```
without luck.

### Expected Behavior

Serve the app normally.

### GitHub Repo

https://github.com/vendramini/nx-docker-test

### Steps to Reproduce

1. Run `docker compose up --build`.

### Nx Report

Node           : 22.0.0
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 10.8.1

nx (global)        : 19.0.0
nx                 : 19.8.4
@nx/js             : 19.8.4
@nx/jest           : 19.8.4
@nx/linter         : 19.8.4
@nx/eslint         : 19.8.4
@nx/workspace      : 19.8.4
@nx/devkit         : 19.8.4
@nx/eslint-plugin  : 19.8.4
@nx/next           : 19.8.4
@nx/react          : 19.8.4
@nrwl/tao          : 19.8.4
@nx/web            : 19.8.4
@nx/webpack        : 19.8.4
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/next/plugin
@nx/jest/plugin
@nx/eslint/plugin

### Failure Logs

nextjs-1  | 2024-10-05T00:27:58: PM2 log: Launching in no daemon mode
nextjs-1  | 2024-10-05T00:27:58: PM2 log: App [nx:0] starting in -fork mode-
nextjs-1  | 2024-10-05T00:27:58: PM2 log: App [nx:0] online
nextjs-1  | (node:76) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
nextjs-1  | (Use `node --trace-deprecation ...` to show where the warning was created)
nextjs-1  |  NX   Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
nextjs-1  | Pass --verbose to see the stacktraces.

### Operating System

- [ ] macOS
- [x] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by AndyBe2001** [comment]
Hi, I have exactly the same issue with Nextjs build process not in Docker but in CI/CD tools like Vercel deployment and Github Action.

For me it work during manual exection by running ``` npx nx build <project-name> ``` or ``` npx nx affected -f build ``` directly on a terminal, it can be a windows or linux terminal (I tried to run it on aws ec2 ubuntu t3.large instance) and it's fine.

I also tryied to host Github Action Runner on the same aws ec2 instance, I also throw the same exact error.

For me this error only happen when building Nextjs project, build of Nestjs project is fine.

For more information, those are what I tried:
- Make sure the project.json file for all project contains name
- Make sure to update all dependencies to the last now nx@19.8.4 and all related @nx/...@19.8.4
- Add ``` run: npx nx reset ``` on github workflow before build
- Run ``` npx nx repair ``` on my local machine then push it
- Use ``` --skip-nx-cache ``` for building process

Still happen to have this bug
